atomic_reactor/dirs.py

```diff
@@ -103,13 +103,11 @@ class RootBuildDir(object):
     +-- x86_64/
     """
 
-    def __init__(self, path: Path, platforms: List[str]) -> None:
+    def __init__(self, path: Path) -> None:
         if not path.exists():
             raise FileNotFoundError(f"Path {path} does not exist.")
         self.path = path
-        if not platforms:
```

Comment:
so if you now won't raise error here, you can just call copy_sources after init, and it won't do anything because you didn't call init_build_dirs, that doesn't seems nice

Reply:
@rcerven 

In this patch, a RootBuildDir object is initialized and passed to a workflow object before initializing the build directories inside with determined platforms, because for the binary container build, the initialization has to be delayed into the check_and_set_platforms plugin.